src/models/scatter.js

```diff
@@ -455,7 +455,7 @@ nv.models.scatter = function() {
                     return 'translate(' + nv.utils.NaNtoZero(x(getX(d[0],d[1]))) + ',' + nv.utils.NaNtoZero(y(getY(d[0],d[1]))) + ')'
                 })
                 .remove();
-            points.filter(function (d) { return scaleDiff || sizeDiff || getDiffs(d, 'x', 'y'); })
+            points.filter(function (d) { return scaleDiff || sizeDiff || getDiffs(d, 'x', 'y') || getDiffs(d, 0, 1); })
```

Comment:
Actually it's the same for "size" and "shape", I think I chose the direct approach because it makes the cache function simpler, i'm going to have to re-think that part a bit to allow for those custom getters.